### PR TITLE
Rebuild staging project if it has support package was staged

### PR DESCRIPTION
**Pull request by nilxam**
related to https://progress.opensuse.org/issues/15448

### CODE REVIEW

File: osc-staging.py [hunk 1]
```diff
@@ -331,14 +331,19 @@ def do_staging(self, subcmd, opts, *args):
 
             if version_openqa == version_totest and not totest_dirty:
                 cmd = AcceptCommand(api)
+                accepted_prjs = []
                 for prj in args[1:]:
-                    if not cmd.perform(api.prj_from_letter(prj), opts.force):
+                    if cmd.perform(api.prj_from_letter(prj), opts.force):
+                        accepted_prjs.append(prj)
```
Comment (DimStar77) [inline osc-staging.py]: I'd likely no go through the indirection here, but do the ```cmd.reset_rebuild_data(accepted_prjs)``` right away for the accepted project.

Consider this (unfortunately still very common) issue: ```osc staging accept C D E F G``` => while accepting G, a failure happens (obs timeout, as so often) -> C D E and F will not be reset in the state; while the person accepting stagings will likely force-accept the remainder of G

Comment (nilxam) [inline osc-staging.py]: fine with me, but that might generate multiple times commits per one time accepting, ie. accepting 5 staging project will committing 5 times. is it ok for you?

File: suppkg_rebuild.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    def get_source_packages(self, project):
+        """Return the list of packages in a project."""
+        query = {'expand': 1}
+        root = ET.parse(http_GET(makeurl(self.apiurl,['source', project],
+                                 query=query))).getroot()
+        packages = [i.get('name') for i in root.findall('entry')]
+        
+        return packages
+
+    def get_prj_pseudometa(self, project):
```
Comment (jberry-suse) [inline suppkg_rebuild.py]: Is there a reason `stagingapi` cannot be used for this and other already available functions?

Comment (jberry-suse) [inline suppkg_rebuild.py]: Is there a reason `stagingapi` cannot be used for this and other already available functions?

File: suppkg_rebuild.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        return files
+
+    def get_staging_projects(self, project):
+        """  
+        Get all letter staging projects ie. without adi
+        """  
+
+        projects = []
+	staging_project = '%s:Staging' % project
+	query = "id?match=starts-with(@name,'{}:')".format(staging_project)
```
Comment (jberry-suse) [inline suppkg_rebuild.py]: nit: indentation

File: suppkg_rebuild.py [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+    def crawl(self):
+        """Main method"""
+        rebuild_data = self.load_rebuild_data(self.project + ':Staging', 'dashboard', 'support_pkg_rebuild')
+        if rebuild_data is None:
+	    print "There is no support_pkg_rebuild file!"
+	    return            
+
+        logging.info('Gathering support package list from %s' % self.project)
+	support_pkgs = self.get_support_package_list(self.project, 'standard')
```
Comment (jberry-suse) [inline suppkg_rebuild.py]: nit: indentation

File: suppkg_rebuild.py [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+        for stg in staging_projects:
+            prj_meta = self.get_prj_pseudometa(stg)
+            prj_staged_packages = [req['package'] for req in prj_meta['requests']]
+            for pkg in support_pkgs:
+                if files.get(pkg) and files.get(pkg) in prj_staged_packages:
+                    if files.get(pkg) not in cand_sources[stg]:
+                        cand_sources[stg].append(files.get(pkg))
+
+	tree = ET.parse(rebuild_data)
+	root = tree.getroot()
```
Comment (jberry-suse) [inline suppkg_rebuild.py]: nit: indentation (2 more instances below)

File: suppkg_rebuild.py [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+                    continue
+                if bname.startswith('::import::'):
+                    continue
+                if result.group(4) == 'src':
+                    continue
+                files[bname] = package.attrib['package']
+
+        return files
+
+    def get_staging_projects(self, project):
```
Comment (jberry-suse) [inline suppkg_rebuild.py]: More code that's in `stagingapi` minus the filter to letter, but that code is also in `get_staging_projects_short()`.

**Comment by lnussel** [comment]
ok now?

**Pull request merged**